core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -2874,16 +2874,61 @@ class KafkaApis(val requestChannel: RequestChannel,
     }
   }
 
-  def handleListClientMetricsResources(request: RequestChannel.Request): Unit = {
-    val listClientMetricsResourcesRequest = request.body[ListClientMetricsResourcesRequest]
+  private def handleListConfigResources(request: RequestChannel.Request): Unit = {
+    val listConfigResourcesRequest = request.body[ListConfigResourcesRequest]
 
     if (!authHelper.authorize(request.context, DESCRIBE_CONFIGS, CLUSTER, CLUSTER_NAME)) {
-      requestHelper.sendMaybeThrottle(request, listClientMetricsResourcesRequest.getErrorResponse(Errors.CLUSTER_AUTHORIZATION_FAILED.exception))
+      requestHelper.sendMaybeThrottle(request, listConfigResourcesRequest.getErrorResponse(Errors.CLUSTER_AUTHORIZATION_FAILED.exception))
     } else {
-      val data = new ListClientMetricsResourcesResponseData().setClientMetricsResources(
-        clientMetricsManager.listClientMetricsResources.stream.map(
-          name => new ClientMetricsResource().setName(name)).collect(Collectors.toList()))
-      requestHelper.sendMaybeThrottle(request, new ListClientMetricsResourcesResponse(data))
+      val data = new ListConfigResourcesResponseData()
+
+      if (request.header.apiVersion() == 0) {
+        // Version 0 only supports client metrics.
+        data.setConfigResources(clientMetricsManager.listClientMetricsResources.stream.map(name =>
+            new ListConfigResourcesResponseData.ConfigResource().setResourceName(name).setResourceType(ConfigResource.Type.CLIENT_METRICS.id)
+          ).collect(Collectors.toList()))
+      } else {
+        // From version 1, supports all ConfigResource.Type.
+        var resourceTypes = listConfigResourcesRequest.data().resourceTypes()
+        if (resourceTypes.isEmpty) {
+          resourceTypes = util.List.of(
```

Comment:
Could we define a constant in `ConfigResource` that includes all of the types so that someone adding a type will not risk missing changing this code?